### PR TITLE
Updated the Model docs - for the ALIGN model

### PR DESCRIPTION
**Pull request by 1himan**
# What does this PR do?

Fixes # (issue)

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#create-a-pull-request),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by 1himan** [comment]
hey @stevhliu can you help me with this one? this is my first time making a PR here 🤗. Hit me up if you need me to make any further updates in the docs.

File: docs/source/en/model_doc/align.md [hunk 1]
```diff
@@ -13,70 +13,182 @@ specific language governing permissions and limitations under the License.
...
+The example below demonstrates how to retrieve zero-shot image labels with [Pipeline] or the [AutoModel] class.
 
-This model was contributed by [Alara Dirik](https://huggingface.co/adirik).
-The original code is not released, this implementation is based on the Kakao Brain implementation based on the original paper.
+<hfoptions id="usage">  
+<hfoption id="Pipeline">
 
-## Usage example
+```py
```
Comment (stevhliu) [inline docs/source/en/model_doc/align.md:37]: Can simplify it a bit to:

```py
import torch
from transformers import pipeline

pipeline = pipeline(
    task="zero-shot-image-classification",
    model="kakaobrain/align-base",
    device=0,
    torch_dtype=torch.bfloat16
)

candidate_labels = [
    "a photo of a dog",
    "a photo of a cat",
    "a photo of a person"
]

pipeline("https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg", candidate_labels=candidate_labels)
```

File: docs/source/en/model_doc/align.md [hunk 2]
```diff
@@ -13,70 +13,182 @@ specific language governing permissions and limitations under the License.
...
+outputs = pipe(image, candidate_labels=candidate_labels)
+print(outputs)
+```
 
-[`AlignProcessor`] wraps [`EfficientNetImageProcessor`] and [`BertTokenizer`] into a single instance to both encode the text and preprocess the images. The following example shows how to get the image-text similarity scores using [`AlignProcessor`] and [`AlignModel`].
+</hfoption>
+<hfoption id="AutoModel">
 
-```python
+```py
```
Comment (stevhliu) [inline docs/source/en/model_doc/align.md:69]: Simplify to:

```py
import torch
import requests
from PIL import Image
from transformers import AutoProcessor, AutoModelForZeroShotImageClassification

processor = AutoProcessor.from_pretrained("kakaobrain/align-base")
model = AutoModelForZeroShotImageClassification.from_pretrained("kakaobrain/align-base").to("cuda")

url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
image = requests.get(url, stream=True)
inputs = Image.open(image.raw).convert("RGB")

image_inputs = processor(images=inputs, return_tensors="pt").to("cuda")
with torch.no_grad():
    image_embeds = model.get_image_features(**image_inputs)

candidate_labels = ["a photo of a dog", "a photo of a cat", "a photo of a person"]
text_inputs = processor(text=candidate_labels, padding=True, return_tensors="pt").to("cuda")
with torch.no_grad():
    text_embeds = model.get_text_features(**text_inputs)

image_embeds = image_embeds / image_embeds.norm(p=2, dim=-1, keepdim=True)
text_embeds  = text_embeds  / text_embeds.norm(p=2, dim=-1, keepdim=True)

logits = (image_embeds @ text_embeds.T) * 100.0
probs  = logits.softmax(dim=-1).cpu().squeeze()

for label, score in zip(candidate_labels, probs):
    print(f"{label:20s} → {score.item():.4f}")
```

File: docs/source/en/model_doc/align.md [hunk 3]
```diff
@@ -13,70 +13,182 @@ specific language governing permissions and limitations under the License.
...
+
+</hfoption>
+<hfoption id="transformers-cli">
+
+```py
+# this command downloads the kakaobrain/align-base for offline use
+transformers-cli download kakaobrain/align-base
+```
+
+</hfoption>
```
Comment (stevhliu) [inline docs/source/en/model_doc/align.md:121]: Can remove this section since its not supported by the `transformers cli`

File: docs/source/en/model_doc/align.md [hunk 4]
```diff
@@ -13,70 +13,182 @@ specific language governing permissions and limitations under the License.
...
+### Quantization
+
+Quantizing `align-base` to 8-bit or 4-bit does not reduce memory usage—in fact, it can increase it. This is because the model is relatively small, and components like the vision encoder aren’t optimized for quantized execution. Quantization benefits are typically seen in larger models (e.g., LLMs) where memory and compute reductions are more significant. Use full precision unless you're experimenting or working under extreme constraints.
+
+---
+
+### Attention Mask Visualization
+
+ALIGN is a **dual‑encoder** (separate image+text) model with **no** autoregressive decoding head, so it does **not** support AttentionMaskVisualizer.  
+That utility only works on models with a `.generate()` step that emits token‑level attentions (e.g. GPT‑style or encoder‑decoder models).
```
Comment (stevhliu) [inline docs/source/en/model_doc/align.md:133]: Can also remove these sections since they aren't relevant

File: docs/source/en/model_doc/align.md [hunk 5]
```diff
@@ -13,70 +13,182 @@ specific language governing permissions and limitations under the License.
...
+
+## Notes
+
+- ALIGN uses a dual-encoder architecture with EfficientNet for images and BERT for text processing
+- The model is designed for zero-shot image-text retrieval tasks
+- Pre-trained on 1.8B image-text pairs from the web
```
Comment (stevhliu) [inline docs/source/en/model_doc/align.md:139]: Already mentioned the architecture and pretraining earlier so no need to include here

```suggestion
- ALIGN projects the text and visual features into latent space and the dot product between the projected image and text features is used as the similarity score. The example below demonstrates how to calculate the image-text similarity score with [`AlignProcessor`] and [`AlignModel`].
```

File: docs/source/en/model_doc/align.md [hunk 6]
```diff
@@ -13,70 +13,182 @@ specific language governing permissions and limitations under the License.
...
+
+```py
+# Example of using ALIGN for image-text similarity
```
Comment (stevhliu) [inline docs/source/en/model_doc/align.md:142]: Indent this code example under the list item above

```py
import torch
import requests
from PIL import Image
from transformers import AlignProcessor, AlignModel

processor = AlignProcessor.from_pretrained("kakaobrain/align-base")
model = AlignModel.from_pretrained("kakaobrain/align-base").to("cuda")

url = "https://huggingface.co/datasets/huggingface/documentation-images/resolve/main/pipeline-cat-chonk.jpeg"
image = requests.get(url, stream=True)
inputs = Image.open(image.raw).convert("RGB")
texts = ["a photo of a cat", "a photo of a dog"]

inputs = processor(images=inputs, text=texts, return_tensors="pt").to("cuda")

with torch.no_grad():
    outputs = model(**inputs)

image_embeds = outputs.image_embeds
text_embeds = outputs.text_embeds

image_embeds = image_embeds / image_embeds.norm(dim=1, keepdim=True)
text_embeds = text_embeds / text_embeds.norm(dim=1, keepdim=True)

similarity_scores = torch.matmul(text_embeds, image_embeds.T)
print("Similarity scores:", similarity_scores)

probs = torch.nn.functional.softmax(similarity_scores, dim=0)
print("Probabilities:", probs)

most_similar_idx = similarity_scores.argmax().item()
print(f"Most similar text: '{texts[most_similar_idx]}'")
```

File: docs/source/en/model_doc/align.md [hunk 7]
```diff
@@ -13,70 +13,182 @@ specific language governing permissions and limitations under the License.
...
+# Get the most similar text
+most_similar_idx = similarity_scores.argmax().item()
+print(f"Most similar text: '{texts[most_similar_idx]}'")
+```
 
 ## AlignConfig
 
-[[autodoc]] AlignConfig
-    - from_text_vision_configs
+[[autodoc]] AlignConfig - from_text_vision_configs
```
Comment (stevhliu) [inline docs/source/en/model_doc/align.md:191]: You don't need to format the `[[autodoc]` here, so you can revert these changes

File: docs/source/en/model_doc/align.md [hunk 8]
```diff
@@ -13,70 +13,182 @@ specific language governing permissions and limitations under the License.
...
-- [Model card](https://huggingface.co/kakaobrain/align-base) of `kakaobrain/align-base` model.
+# Convert to probabilities
+probs = torch.nn.functional.softmax(similarity_scores, dim=0)
+print("Probabilities:", probs)
 
-If you're interested in submitting a resource to be included here, please feel free to open a Pull Request and we will review it. The resource should ideally demonstrate something new instead of duplicating an existing resource.
+# Get the most similar text
+most_similar_idx = similarity_scores.argmax().item()
+print(f"Most similar text: '{texts[most_similar_idx]}'")
+```
```
Comment (stevhliu) [inline docs/source/en/model_doc/align.md:187]: Add a `## Resource` section under here for the blog post.

```md
## Resources

- Refer to the [Kakao Brain’s Open Source ViT, ALIGN, and the New COYO Text-Image Dataset](Kakao Brain’s Open Source ViT, ALIGN, and the New COYO Text-Image Dataset) blog post for more details.
```

**Review by stevhliu — COMMENTED** [review]
Congrats on your first PR to Hugging Face and really nice work! 🤗

File: docs/source/en/model_doc/align.md (hunk 8 above)
Comment (1himan) [reply docs/source/en/model_doc/align.md]: done.

**Comment by 1himan** [comment]
Anything else? :)

File: docs/source/en/model_doc/align.md (hunk 6 above)
Comment (stevhliu) [reply docs/source/en/model_doc/align.md]: Not resolved yet

File: docs/source/en/model_doc/align.md (hunk 7 above)
Comment (stevhliu) [reply docs/source/en/model_doc/align.md:155]: Changes to the `[[autodoc]]` haven't been reverted yet

**Comment by 1himan** [comment]
Now yeah, I think it looks good.

**Review by stevhliu — COMMENTED** [review]
Still a few minor formatting things!

**Review by stevhliu — APPROVED** [review]
Thanks!
